README


This archive contains the distribution Dist-Zilla-Plugin-Test-EOL,
version 0.16:

  Author tests making sure correct line endings are used

This software is copyright (c) 2010 by Florian Ragwitz <rafl@debian.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.025.



Changes
Revision history for Dist-Zilla-Plugin-Test-EOL

0.16    2014-11-17 03:00:59Z
        - allow tests to pass even while upstream modules have warnings under
          perl 5.21.6

0.15    2014-10-31 06:09:12Z
        - use the 'deprecated' warning category to issue the warning in
          [EOLTests]

0.14    2014-10-23 01:37:00Z
        - change default filename from test-eol.t to eol.t
        - add metadata for [MetaConfig]
        - fix missing test prerequisite
        - formally deprecated [EOLTests] (it will continue to exist through
          April 2015).

0.13    2014-10-22 05:44:36Z
        - Dist::Zilla::Plugin::EOLTests and Dist::Zilla::Plugin::Test::EOL are
          now combined into one distribution, Dist-Zilla-Plugin-Test-EOL
        - add "filename" configuration option
        - Test::EOL is injected as a develop prerequisite

0.12    2014-04-10
        - readd Test::EOL prereq that was accidentally removed when moving to
          authorbundle

0.11    2014-04-10
        - improve docs
        - move to authorbundle

0.10    2013-12-18
        - forgot to update encoding and contributors

0.09    2013-12-18
        - rerelease because of DZP::Test::Compile problem on windows

0.08    2013-09-07
        - revert change added in 0.06, Dist::Zilla should now be handling this
          correctly, update your Dist::Zilla if you're having a problem. If you're
          still having a problem after that let me know

0.07    2012-04-04
        - update my other plugins which should also update test deps
        - add generated by comment to test output

0.06    2012-04-03
        - attempt to fix issue where Test::EOL is added to prereqs of consuming
          dists

0.05    2012-04-02
        - add Test::EOL dep ( dolmen )
        - use Test::EOL in generated test ( dolmen )
        - fix bugtracker url
        - add a simple test to check that the test compiles

0.04    2012-01-09
        - Give RAFL more credit

0.03    2011-12-11
        - Fork from Dist::Zilla::Plugin::EOLTests
        - make an author test
        - new standard ::Test namespace
        - use Test::Requires

0.02    2010-06-03
        - Allow checking for trailing whitespace.
          This is enabled by default, but can be turned off with
          trailing_whitespace = 0.

0.01    2010-05-29
        - Split off of Dist::Zilla::Plugin::SanityTests.


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-Test-EOL was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.31.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Author tests making sure correct line endings are used",
  "AUTHOR" => "Florian Ragwitz <rafl\@debian.org>, Caleb Cushing <xenoterracide\@gmail.com>, Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-EOL",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Test::EOL",
  "PREREQ_PM" => {
    "Data::Section" => "0.004",
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "Sub::Exporter::ForMethods" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::EOL" => 0,
    "Test::More" => 0,
    "Test::Warnings" => "0.009"
  },
  "VERSION" => "0.16",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Data::Section" => "0.004",
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Sub::Exporter::ForMethods" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::EOL" => 0,
  "Test::More" => 0,
  "Test::Warnings" => "0.009",
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-Test-EOL was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-Test-EOL was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.006;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.025.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.006"
      },
      "dist_abstract" => "Author tests making sure correct line endings are used",
      "dist_author" => [
        "Florian Ragwitz <rafl\@debian.org>",
        "Caleb Cushing <xenoterracide\@gmail.com>",
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-Test-EOL",
      "dist_version" => "0.16",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::Test::EOL",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Data::Section" => "0.004",
        "Dist::Zilla::File::InMemory" => 0,
        "Dist::Zilla::Role::FileGatherer" => 0,
        "Dist::Zilla::Role::PrereqSource" => 0,
        "Dist::Zilla::Role::TextTemplate" => 0,
        "Moose" => 0,
        "Sub::Exporter::ForMethods" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::pushd" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::EOL" => 0,
        "Test::More" => 0,
        "Test::Warnings" => "0.009",
        "perl" => "5.006"
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::pushd" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::EOL" => 0,
      "Test::More" => 0,
      "Test::Warnings" => "0.009",
      "perl" => "5.006"
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
